src/Streams.Core/Streams.fs

```diff
@@ -289,6 +289,36 @@ module Stream =
             { new Iterable with 
                  member __.Bulk() = bulk() 
                  member __.Iterator = iterator()  })
+
+    /// <summary>Produces an infinite stream from generator function.</summary>
+    /// <param name="generator">A function used to generate values.</param>
+    /// <returns>The result stream.</returns>
+    let infinite (generator : Unit -> 'T) : Stream<'T> =
```

Comment:
"infinite" is not a great for this.  

- First, all initialization functions would be "initXXX".  Second, I'd recommend reserving a name like "infinite" for a fundamentally functional combinator. 

- Second, on the whole the F# Seq library avoids imperatively generating combinators.  Instead, there's an expectation that imperative sequence generation is done with ``seq { ... }`` expressions (which you are comparing to of course) which are very expressive and capture more variations on imperative generation than is easily done through combinators. That said, I'm not fundamentally opposed to imperatively-generating combinators. But I think all such combinators should have a common adjective in their name.  For example, please consider using the name ``generateInfinite``.

BTW there are some related runtime helpers in the FSharp.Core library here: https://github.com/fsharp/fsharp/blob/master/src/fsharp/FSharp.Core/seq.fsi#L1016 however these are not really for direct use from F# code.

Thanks
Don